sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -355,16 +355,16 @@ def split_next(self):
 
         self.n_nodes += 2
 
-        if self.max_depth is not None and depth == self.max_depth:
+        if (self.max_leaf_nodes is not None
```

Comment:
Diff might be confusing but all I did was check for max_leaf_nodes *before* checking for max_depth

Reply:
This reordered was noticed ;)